File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        r = len(matrix)
        c = len(matrix[0])
        zero_row = [0] * r
        zero_col = [0] * c

        for i in range(r):
            for j in range(c):
                if (matrix[i][j] == 0):
                    zero_row[i] = 1
                    zero_col[j] = 1

        for i in range(r):
            for j in range(c):
                if (zero_row[i] == 1 or zero_col[j] == 1):
                    matrix[i][j] = 0
        

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int r = matrix.size(), c = matrix[0].size();
        vector<int> zeroRow(r, 0), zeroCol(c, 0);

        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (!matrix[i][j]) {
                    zeroRow[i] = 1;
                    zeroCol[j] = 1;
                }
            }
        }

        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (zeroRow[i] || zeroCol[j])
                    matrix[i][j] = 0;
            }
        }
    }
};